### PR TITLE
fix parsing DISTRO from 'content' file (bsc#1122003)

### PR DESCRIPTION
**Pull request by wfeldt**
There might be more than one comma (",") in the string. Assume the cpe string and the product
name might both contain commas.

> Note:
> The CPE string specs were guessed from the examples at http://cpe.mitre.org as the specs
> can't currently be downloaded due to the U.S. government shutdown.

### CODE REVIEW

**Comment by wfeldt** [comment]
Nice test case. :-(

**Comment by srinidhibs** [comment]
> There might be more than one comma (",') in the string. Assume
> the product name never contains one.
> 

Actually, just now read OBS code responsible for generating the DISTRO string (bs_productconvert):

https://github.com/openSUSE/open-build-service/blob/master/src/backend/bs_productconvert#L365-L371

and realized that the RHS of the comma is actually the "summary" string of the product and not the "product name". And checking product.xsd file that is used to validate the product definition, doesn't define any constraints:

```xml
      <xs:element name="summary" type="xs:string">
        <xs:annotation>
          <xs:documentation xml:lang="en">
            Short (one line only!) description of the product.
            Used in the media.1/media file.
          </xs:documentation>
        </xs:annotation>
      </xs:element>
```

In other words, you are right. Anything can be mentioned in "summary" string (including ",") and that will be part of the DISTRO.

> Alternatively we could parse the cpe value correctly but this seems
> like overengineering at this point.

At least, for products defined or created in OBS, this should be fixed in bs_productconvert. Instead of appending the summary string verbatim in the DISTRO entry, bs_productconvert should call `escapeProvides()` method to escape the summary screen. But would YaST / Ruby have trouble in parsing the escaped string? The code is available here:

https://github.com/openSUSE/open-build-service/blob/master/src/backend/bs_productconvert#L890

File: src/include/instserver/routines.rb [hunk 1]
```diff
@@ -25,6 +25,20 @@ def ReadMediaFile(media)
...
+    # @return [Array<String,String>]
+    #
+    # The distro arg looks like
+    #   "cpe:/o:suse:sles:12:sp3,SUSE Linux Enterprise Server 12 SP3"
+    #   "cpe:/o:novell,inc:sles:12:sp3,SUSE Linux Enterprise Server 12 SP3"
+    #
+    def distro_split(distro)
+      x = distro.rpartition(',')
+
+      x[1].empty? ? [x[2], x[1]] : [x[0], x[2]]
```
Comment (srinidhibs) [inline src/include/instserver/routines.rb]: I did a quick test in `irb` and used negative look-ahead regex to match comma **not** followed by a whitespace:

```ruby
irb(main):022:0> def distro_split(distro)
irb(main):023:1> x = distro.rpartition(/,(?! )/)
irb(main):024:1> x[1].empty? ? [x[2], x[1]] : [x[0], x[2]]
irb(main):025:1> end
=> :distro_split
irb(main):026:0> s2 = "cpe:/o:suse:sles:12,SLES, Mini Edition"
=> "cpe:/o:suse:sles:12,SLES, Mini Edition"
irb(main):027:0> cpeid, name = distro_split(s2)
=> ["cpe:/o:suse:sles:12", "SLES, Mini Edition"]
irb(main):028:0>
```
Yes, this will fail if someone places a "whitespace" at the beginning of the summary for the product. But I believe, that should be handled in OBS when `bs_productconvert` generates the `DISTRO` string.
